frontend/src/v5/ui/controls/favouriteCheckbox/favouriteCheckbox.component.tsx

```diff
@@ -0,0 +1,35 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import StarIcon from '@assets/icons/star.svg';
+import { Checkbox } from './favouriteCheckbox.styles';
+
+type IFavouriteCheckbox = {
+	checked: boolean;
+	onChange: (value: boolean) => void;
+	disabled?: boolean;
+	className?: string;
+};
```

Comment:
We can change this to cover all of Checkbox props
```suggestion
type IFavouriteCheckbox = Omit<Omit<CheckboxProps, 'icon'>, 'checkedIcon'>;
```